src/topp/GNPSExport.cpp

```diff
@@ -441,88 +463,61 @@ class TOPPGNPSExport : public TOPPBase
 
           MSExperiment exp;
           specs_list[num_msmaps_cached] = exp;
-
           mzml_files[num_msmaps_cached].load(mzml_file_paths[map_index], specs_list[num_msmaps_cached]);
 
           msmaps_cached[map_index] = num_msmaps_cached;
           num_msmaps_cached += 1;
         }
-
-        MSExperiment::SpectrumType specs = specs_list[msmaps_cached[map_index]][spec_index];
-
-        if (output_file.is_open()) 
-        {
-          // write output
-          writeMSMSBlockHeader_(
-            output_file,
-            output_type,
-            (cons_i + 1),
-            feature.getUniqueId(),
-            charge,
-            feature.getMZ(),
-            spec_index,
-            specs.getRT()
-          );
-
-          output_file << fixed << setprecision(4);
-          for (auto spec_iter = specs.begin(); spec_iter != specs.end(); ++spec_iter)
-          {
-            if ((int) spec_iter->getIntensity() > 0)
-            {
-              output_file << spec_iter->getMZ() << "\t" << (int) spec_iter->getIntensity() << "\n";
-            }
-          }
-
-          output_file << "END IONS" << "\n" << endl;
-        }
       }
-      else if (output_type == "merged_spectra")
-      {
-        // discard poorer precursor spectra for 'merged_spectra' and 'full_spectra' output
-        if (pept_cutoff != -1 && pepts.size() > (unsigned long) pept_cutoff)
-        {
-          pepts.erase(pepts.begin()+pept_cutoff, pepts.end());
-        }
-        for (vector<pair<int,int>>::iterator pepts_iter=pepts.begin(); pepts_iter!=pepts.end(); pepts_iter++)
-        {
-          int map_index = pepts_iter->first;
-
-          // load missing MzMLFile files + MSExperiment specs
-          if (msmaps_cached.find(map_index) == msmaps_cached.end())
-          {
-            MzMLFile mzmlfile;
-            mzml_files[num_msmaps_cached] = mzmlfile;
-
-            MSExperiment exp;
-            specs_list[num_msmaps_cached] = exp;
-
-            mzml_files[num_msmaps_cached].load(mzml_file_paths[map_index], specs_list[num_msmaps_cached]);
 
-            msmaps_cached[map_index] = num_msmaps_cached;
-            num_msmaps_cached += 1;
-          }
-        }
-
-        int best_mapi = pepts[0].first;
-        int best_speci = pepts[0].second;
-        auto best_spec = specs_list[msmaps_cached[best_mapi]][best_speci];
+      // identify most intense spectrum
+      const int best_mapi = (pepts[0]).first;
+      const int best_speci = (pepts[0]).second;
+      auto best_spec = specs_list[msmaps_cached[best_mapi]][best_speci];
+
+      // write block output header
+      writeMSMSBlockHeader_(
+        output_file,
+        output_type,
+        (cons_i + 1),
+        feature.getUniqueId(),
+        charge,
+        feature.getMZ(),
+        best_speci,
+        best_spec.getRT()
+      );
+
+      // store outputted spectra in MSExperiment
+      MSExperiment exp;
+
+      // add most intense spectrum to MSExperiment
+      exp.addSpectrum(best_spec);
```

Comment:
I'm unsure whether this line (without logic check that output_type!= 'merged_spectra') causes the most intense spectrum to be added twice (due to cosine similarity of 1 to itself in the 'merged_spectra' branch). And if so, would that make a difference after flattening and binning for reported intensities?